### ISSUE TITLE
Why is 'prompt' a global boolean flag? Is it a bug?

### ISSUE THREAD

**Issue by emilyst** [mhinz/vim-grepper#154]
In `vim-grepper`'s docs, the `g:grepper.prompt` flag is documented as a global flag with a default value of `1`. There is no real explanation there about what this flag does or why it exists.

<img width="590" alt="capture 2018-03-25 at 16 26 10" src="https://user-images.githubusercontent.com/528994/37881213-695f496c-3049-11e8-9a9e-ab5813678d97.png">

Because using `:Grepper` is a two-step process, I tried setting `g:grepper.prompt` to `0` to see if I could cause `:Grepper` to become a one-step command without having to type one of the explicit tool-based commands (e.g., `:GrepperRg`). Easier to tab-complete, since the tool-based commands pollute the command namespace, shorter, and would always use the default tool.

However, though this flag's setting does indeed skip the prompt, it never populates the tool's query. I've tried it with multiple tools. When I run the command as `:verbose :Grepper active` (I used "active" as the query for this example), I see output that leads me to believe that `vim-grepper` is interpreting the query as a flag and ignoring it.

```
Ignore unknown flag: active
grepper: running ['sh', '-c', 'ag --vimgrep ']
```

It seems as if `g:grepper.prompt` should probably not be a global flag for configuration. I can only find it used to control `:Grepper` calls to prevent it from prompting when its behavior is being overridden for specific tools or situations (such as the various tool commands defined at the end of the plugin script).

**Comment by mhinz** [comment]
`:Grepper` is the lower-level command every other command is based on. It has flags that can overwrite all options set in the vimrc on runtime. See `:h :Grepper`.

In fact, you can have a look at how the other generated commands work:

```
:com GrepperGit
    Name        Args       Address   Complete  Definition
    GrepperGit  +                    file          Grepper -noprompt -tool git -query <args>
```

So, when you set`g:grepper.prompt = 0`, `:Grepper` is effectively running `:Grepper -noprompt`. Since you didn't provide a query with `-query`, the query is empty. When the query is empty, it tries to looking up the cword (the word under the cursor) if there is one.

Long story short, you want `:Grepper -query active`.

If that's too long for your purposes, create your own custom command based on `:Grepper` or use an abbreviation like `:cabbrev grepper Grepper -query`.

**Comment by emilyst** [comment]
> `:Grepper` is the lower-level command every other command is based on. 

It seems like, from the usage in the README, you envisage people using this command primarily?

> It has flags that can overwrite all options set in the vimrc on runtime. See `:h :Grepper`.

I have been referring to the docs, so I have, in fact, seen this.

> In fact, you can have a look at how the other generated commands work:

I did see these.

> Long story short, you want `:Grepper -query active`.

I do not want this.

> If that's too long for your purposes, create your own custom command based on `:Grepper` or use an abbreviation like `:cabbrev grepper Grepper -query`.

I have already done this. I am not asking how to accomplish a shorter command. I found what seems like a strange behavior while creating my own shorter command.

* * *

> So, when you set `g:grepper.prompt = 0`, `:Grepper` is effectively running `:Grepper -noprompt`. Since you didn't provide a query with `-query`, the query is empty. When the query is empty, it tries to looking up the cword (the word under the cursor) if there is one.

Right, so, this is the crux of my question, I suppose. If there's a *global* flag which *requires* that each invocation of the command include a local flag, doesn't that mean there is something iffy about that global flag? Or that it requires some explanation in the docs that this behavior would result?

**Comment by mhinz** [comment]
The local flag is _not_ required, but I added a small note to `:h g:grepper.prompt` that hopefully makes the behaviour more obvious.

Before the `g:grepper.operator` namespace was introduced, `g:grepper.prompt` was the only way to disable the prompt for mappings that use `<plug>(GrepperOperator)`. Nowadays the only difference between the global and the operator namespace is that the global namespace uses `prompt = 1` and and the operator namespace `prompt = 0` by default. So there's usually no use for `g:grepper.prompt`.

**Comment by mhinz** [comment]
I understand what you mean and I think the new note in `g:grepper.prompt` addresses that confusion, so I'm closing this.

Thanks for bringing this up! ✨
